Support ParameterRef-less GatewayClass

**Please describe the problem you have**
`gatewayclasses.spec.parametersRef` is optional and passes validation as such. However, the operator panics for this use case:

```
2021-04-30T18:06:17.101Z	INFO	gatewayclass_controller	synced status for gatewayclass	{"name": "sample-gatewayclass"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1369396]

goroutine 417 [running]:
github.com/projectcontour/contour-operator/internal/objects/gateway.ContourForGateway(0x1a6d918, 0xc00084f740, 0x1a7fc70, 0xc0002b3860, 0xc0000b2ea0, 0x0, 0x0, 0xc00083fb80)
	/internal/objects/gateway/gateway.go:81 +0x96
github.com/projectcontour/contour-operator/internal/operator/controller/gateway.(*reconciler).ensureGateway(0xc00013cd40, 0x1a6d918, 0xc00084f740, 0xc0000b2ea0, 0xc0000b2ea0, 0x0)
	/internal/operator/controller/gateway/controller.go:201 +0x9b
github.com/projectcontour/contour-operator/internal/operator/controller/gateway.(*reconciler).Reconcile(0xc00013cd40, 0x1a6d918, 0xc00084f740, 0xc000753e20, 0xe, 0xc000753e10, 0x7, 0xc00084f740, 0xc000030000, 0x172fc60, ...)
	/internal/operator/controller/gateway/controller.go:146 +0x3b6
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0002ce5a0, 0x1a6d870, 0xc00013ce00, 0x16f51c0, 0xc000852120)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.0/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0002ce5a0, 0x1a6d870, 0xc00013ce00, 0xc000577500)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.0/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc0007523e0, 0xc0002ce5a0, 0x1a6d870, 0xc00013ce00)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.0/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.0-beta.0/pkg/internal/controller/controller.go:210 +0x425
```

--- a/pkg/validation/validation.go
+++ b/pkg/validation/validation.go
@@ -159,8 +159,8 @@ func Gateway(ctx context.Context, cli client.Client, gw *gatewayv1alpha1.Gateway
 		}
 	}
 	if !admitted {
-		errs = append(errs, fmt.Errorf("invalid gatewayclass %s; status must be %s=%s", gc.Name,
-			gatewayv1alpha1.GatewayClassConditionStatusAdmitted, metav1.ConditionTrue))
+		return nil, fmt.Errorf("invalid gatewayclass %s; status must be %s=%s", gc.Name,
+			gatewayv1alpha1.GatewayClassConditionStatusAdmitted, metav1.ConditionTrue)
 	}
 
 	if err := gatewayListeners(gw); err != nil {

--- a/pkg/validation/validation_test.go
+++ b/pkg/validation/validation_test.go
@@ -913,6 +913,53 @@ func TestGateway(t *testing.T) {
 			},
 			expected: false,
 		},
+		"invalid gatewayclass no parameters ref": {
+			contour: &operatorv1alpha1.Contour{
+				TypeMeta: metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-gatewayclass-no-parameters-ref-contour",
+				},
+				Spec: operatorv1alpha1.ContourSpec{
+					Namespace: operatorv1alpha1.NamespaceSpec{
+						Name: ns.Name,
+					},
+					GatewayClassRef: pointer.StringPtr("invalid-gatewayclass-no-parameters-ref-gc"),
+				},
+			},
+			gc: &gatewayv1alpha1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "invalid-gatewayclass-no-parameters-ref-gc",
+				},
+				Spec: gatewayv1alpha1.GatewayClassSpec{
+					Controller: operatorv1alpha1.GatewayClassControllerRef,
+				},
+				Status: gatewayv1alpha1.GatewayClassStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   string(gatewayv1alpha1.GatewayClassConditionStatusAdmitted),
+							Status: metav1.ConditionFalse,
+						},
+					},
+				},
+			},
+			gateway: &gatewayv1alpha1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      "invalid-gatewayclass-no-parameters-ref-gateway",
+				},
+				Spec: gatewayv1alpha1.GatewaySpec{
+					GatewayClassName: "invalid-gatewayclass-no-parameters-ref-gc",
+					Listeners: []gatewayv1alpha1.Listener{
+						{
+							Port:     gatewayv1alpha1.PortNumber(int32(1)),
+							Protocol: gatewayv1alpha1.HTTPProtocolType,
+						},
+					},
+				},
+			},
+			expected: false,
+		},
 		"invalid contour spec ns": {
 			contour: &operatorv1alpha1.Contour{
 				TypeMeta: metav1.TypeMeta{},

